dynamiqs/utils/utils.py

```diff
@@ -412,17 +414,60 @@ def expect(O: Tensor, x: Tensor) -> Tensor:
         is real. One can then keep only the real values of the returned tensor using
         `dq.expect(O, x).real`.
 
-    Warning:
-        This function does not yet support tensors in the bra format.
-
     Args:
         O (n, n): Arbitrary operator.
-        x (..., n, 1) or (..., n, n): Ket or density matrix.
+        x (..., n, 1) or (..., 1, n) or (..., n, n): Ket, bra or density matrix.
 
     Returns:
         (...): Complex-valued expectation value.
+
+    Raises:
+        TypeError: If the input tensor is not a ket, bra or density matrix.
```

Comment:
```suggestion
        ValueError: If the input tensor is not a ket, bra or density matrix.
```
All exceptions in this file should be `ValueError` I think.